File: src/Modules/Learning/UniHub.Learning.Domain/Documents/DocumentStatus.cs
namespace UniHub.Learning.Domain.Documents;

/// <summary>
/// Trạng thái phê duyệt của tài liệu
/// </summary>
public enum DocumentStatus
{
    /// <summary>
    /// Chưa gửi duyệt - Draft
    /// </summary>
    Draft = 1,
    
    /// <summary>
    /// Đang chờ phê duyệt (Pending Approval)
    /// </summary>
    PendingApproval = 2,
    
    /// <summary>
    /// Đã được duyệt và xuất bản (Approved)
    /// </summary>
    Approved = 3,
    
    /// <summary>
    /// Bị từ chối (Rejected)
    /// </summary>
    Rejected = 4,
    
    /// <summary>
    /// Đã bị xóa
    /// </summary>
    Deleted = 5
}

File: frontend/src/features/learning/lib/documentStatus.ts
import type { TFunction } from 'i18next'

export type DocumentStatusTone = 'success' | 'danger' | 'warning' | 'muted'

const STATUS_LABEL_KEYS: Record<string, string> = {
  Draft: 'learning.documentStatus.draft',
  PendingApproval: 'learning.documentStatus.pendingApproval',
  Approved: 'learning.documentStatus.approved',
  Rejected: 'learning.documentStatus.rejected',
  Deleted: 'learning.documentStatus.deleted',
}

export function getDocumentStatusLabel(status: string, t: TFunction): string {
  const key = STATUS_LABEL_KEYS[status]
  return key ? t(key) : status
}

export function getDocumentStatusTone(status: string): DocumentStatusTone {
  if (status === 'Approved') return 'success'
  if (status === 'Rejected') return 'danger'
  if (status === 'Deleted') return 'muted'
  return 'warning'
}

export function documentStatusBadgeClass(tone: DocumentStatusTone): string {
  switch (tone) {
    case 'success':
      return 'text-emerald-700'
    case 'danger':
      return 'text-rose-700'
    case 'muted':
      return 'text-slate-500'
    default:
      return 'text-amber-700'
  }
}
